File: ICAS_NonRPA/Python/DeleteFile/DeleteFile.py
'''
Copyright 2020 Infosys Ltd.
Use of this source code is governed by Apache 2.0 license that can be found in the LICENSE file or at 
https://opensource.org/licenses/Apache-2.0 .
'''
# Auto generated - on: 20190723, by: akhil.krishna
# ------------------------------------------------------------------------------------------
# Variables and Libraries
import sys, re, os
# from Ops_Conn import *
from abstract_bot import Bot
# ------------------------------------------------------------------------------------------
# Arguments: ["string","file_path"]
# deletes a file
# ------------------------------------------------------------------------------------------
class DeleteFile(Bot):


  def bot_init(self):
    pass


  def execute(self, executeContext):
    # self.previous_status,self.arg_dict = readArguments('Script expects arguments')
    # if responseprevious_status != 'OK':
      # checkExit(1)
    if "file_path" in executeContext:
      file_path = executeContext["file_path"]
    else:
      return {'Missing argument': 'file_path'}
    try:
      os.remove(file_path)

      return {'status' : 'success'}
    except Exception as e:
      return {'Exception' : str(e)}

    

if __name__ == '__main__':
    context = {}
    output = {}
    obj_snow = DeleteFile()

    context = {'file_path':'D:\\Dummy\\delete\\new.txt'}

    # print(dir(obj_snow))
    obj_snow.bot_init()
    output = obj_snow.execute(context)
    print(output)
    # write context to json

File: ICAS_NonRPA/Java/DeleteFile/DeleteFile.java
/*
Copyright 2020 Infosys Ltd.
Use of this source code is governed by Apache 2.0 license that can be found in the LICENSE file or at 
https://opensource.org/licenses/Apache-2.0 .
*/
package com.infosys.impact.botfactory.microbots.file;

import static com.infosys.impact.botfactory.microbots.framework.Validation.EXISTS;
import static com.infosys.impact.botfactory.microbots.framework.Validation.FILE;
import static com.infosys.impact.botfactory.microbots.framework.Validation.FOLDER;

import java.io.File;
import java.io.IOException;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.infosys.impact.botfactory.microbots.framework.Error;
import com.infosys.impact.botfactory.microbots.framework.Errors;
import com.infosys.impact.botfactory.microbots.framework.InputParameter;
import com.infosys.impact.botfactory.microbots.framework.Status;

public class DeleteFile {

	private static final Logger log = LoggerFactory.getLogger(DeleteFile.class);
	@Errors(exceptions= {
			@Error(errorCode="0102030405", errorMessagePattern ="MISC_IO_EXCEPTION" , exceptionClass=IOException.class)
		})
	public Status execute(
			@InputParameter(name="SourcePath", constraints= {FOLDER, EXISTS}) String sourcePath,
			@InputParameter(name="FileName", constraints= {FILE, EXISTS}) String fileName
//			@OutputParameter(name="Files") ObjectHolder<List<String>> files
			
			) throws Exception {

		File file = new File(sourcePath, fileName);

		try {

			if (file.delete()) {

				log.info("**********************************************\n");
				log.info(fileName + " File deleted successfully");
				log.info("\n**********************************************");
				

			} else {
				log.info("Failed to delete the file");
			}
			
			return new Status("00","File deleted Successfully from ");
		} catch (Exception e) {

			String className = this.getClass().getSimpleName();
			log.info("Display Error code ::" + className);
			return new Status("Error",e.getMessage(),e);

		}

	}

}
